vault/init.sls

```diff
@@ -8,13 +8,26 @@ vault packages:
 
 download vault:
   cmd.run:
-    - name: curl --silent -L https://releases.hashicorp.com/vault/{{ vault.version }}/vault_{{ vault.version }}_linux_amd64.zip -o /tmp/vault.zip
-    - unless: test -e /tmp/vault.zip
+    - name: curl --silent -L https://releases.hashicorp.com/vault/{{ vault.version }}/vault_{{ vault.version }}_linux_amd64.zip -o /tmp/vault_{{ vault.version }}_linux_amd64.zip
+    - unless: test -e /tmp/vault_{{ vault.version }}_linux_amd64.zip
+
+download sha256sums vault:
+  cmd.run:
+    - name: curl --silent -L https://releases.hashicorp.com/vault/{{ vault.version }}/vault_{{ vault.version }}_SHA256SUMS -o /tmp/vault_{{ vault.version }}_SHA256SUMS
+    - unless: test -e /tmp/vault_{{ vault.version }}_SHA256SUMS
+
+check sha256sum vault:
+  cmd.wait:
+    - name: sha256sum -c vault_{{ vault.version }}_SHA256SUMS 2>&1 | grep OK
```

Comment:
what program provides this executable? we should ensure the package is installed if its not provided by the base OS

Also can you verify that testing for the exit code does not work? testing for a case sensitive string doesnt appeal to me as `NOT OK` would cause this to pass along with a multitude of other things with two letters in it